pkg/models/migration/legacy/to.go

```diff
@@ -160,11 +160,14 @@ func ToLegacyStorageSpec(storage *models.SpecConfig) (model.StorageSpec, error)
 			URL:           storage.Params["URL"].(string),
 		}, nil
 	case models.StorageSourceLocalDirectory:
-		return model.StorageSpec{
+		storageSpec := model.StorageSpec{
 			StorageSource: model.StorageSourceLocalDirectory,
 			Path:          storage.Params["SourcePath"].(string),
-			ReadWrite:     storage.Params["ReadWrite"].(bool),
-		}, nil
+		}
+		if storage.Params["ReadWrite"] != nil {
+			storageSpec.ReadWrite = storage.Params["ReadWrite"].(bool)
```

Comment:
It might be safer here to use the two value type assertion before the cast, to make sure the underlying interface is a bool. At the moment if the interface{} in storage.Params["ReadWrite"] is somehow a string, it'll panic.

```
if b, ok := storage.Params["ReadWrite"].(bool); ok {
    storageSpec.ReadWrite = b 
}
```

Reply:
fixed in https://github.com/bacalhau-project/bacalhau/pull/2900/commits/f4d2598eca6de26e4bdbafecddd8aaf9e31c734d